driver/driver.go

```diff
@@ -3,51 +3,85 @@
 package driver
 
 import (
+	"crypto/tls"
 	"database/sql"
 	sqldriver "database/sql/driver"
 	"fmt"
 	"io"
-	"strings"
+	"net/url"
+	"regexp"
 
 	"github.com/go-mysql-org/go-mysql/client"
 	"github.com/go-mysql-org/go-mysql/mysql"
 	"github.com/pingcap/errors"
 	"github.com/siddontang/go/hack"
 )
 
+// Map of dsn address (makes more sense than full dsn?) to tls Config
+var customTLSConfigMap = make(map[string]*tls.Config)
+
 type driver struct {
 }
 
-// Open: DSN user:password@addr[?db]
+// Open: DSN
+// Support both legacy style DSNs: user:password@addr[?db]
+// And more standard DSNs: user:password@addr/db?param=value
+// Optional parameters are supported in the standard DSN
 func (d driver) Open(dsn string) (sqldriver.Conn, error) {
-	lastIndex := strings.LastIndex(dsn, "@")
-	seps := []string{dsn[:lastIndex], dsn[lastIndex+1:]}
-	if len(seps) != 2 {
-		return nil, errors.Errorf("invalid dsn, must user:password@addr[?db]")
+	// If a "/" occurs after "@" and then no more "@" or "/" occur after that
+	standardDSN, matchErr := regexp.MatchString("@[^@]+/[^@/]+", dsn)
```

Comment:
Can you add an unit test to explain this behaviour? That's more easier to understand. It's OK that you don't want to add it.

Reply:
Added a [test](https://github.com/atomicules/go-mysql/blob/a5b62b18e3fd0f43060a0a6a3ba9cf96de6f142a/driver/driver_test.go#L84-L107). It required me to split out the [parsing into a separate function](https://github.com/atomicules/go-mysql/blob/a5b62b18e3fd0f43060a0a6a3ba9cf96de6f142a/driver/driver.go#L45-L79). See commit: https://github.com/go-mysql-org/go-mysql/pull/673/commits/a5b62b18e3fd0f43060a0a6a3ba9cf96de6f142a